.github/workflows/python-package.yml

```diff
@@ -4,45 +4,41 @@ name: Python package
 
 on:
   push:
-    branches: [ main ]
+    branches: [main]
   pull_request:
-    branches: [ main ]
+    branches: [main]
 
 defaults:
   run:
     shell: bash
-    working-directory: dataquality
 
 jobs:
   build:
     runs-on: ubuntu-latest
-    strategy:
-      matrix:
-        # TODO add 3.8 when this bug is fixed upstream: https://github.com/great-expectations/great_expectations/issues/1651
-        python-version: [3.6, 3.7]
     steps:
-    - uses: actions/checkout@v2
-    - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v2
-      with:
-        python-version: ${{ matrix.python-version }}
-    - name: Install dev dependencies
-      run: |
-        python -m pip install --upgrade pip
-        pip install -r requirements-dev.txt
-# TODO dig up consistency fix for isort
-#    - name: Check linting on imports using isort
-#      run: |
-#        isort --check-only */*.py
-    - name: Check linting with black
-      run: |
-        black --check .
-    - name: Typecheck with mypy
-      run: |
-        mypy dataquality/
-    - name: Install package
-      run: |
-        pip install -e .
-    - name: Test with pytest
-      run: |
-        pytest tests
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          # TODO add 3.8 when this bug is fixed upstream: https://github.com/great-expectations/great_expectations/issues/1651
+          python-version: 3.7
+      - name: Install dev dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements-dev.txt
+      # TODO dig up consistency fix for isort
+      #    - name: Check linting on imports using isort
+      #      run: |
+      #        isort --check-only */*.py
+      - name: Check linting with black
+        run: |
+          black --check .
+      - name: Typecheck with mypy
+        run: |
+          mypy src/
+      - name: Install package
+        run: |
+          pip install -e .
```

Comment:
We could to d the testing in a docker image stage and recapture the install caching rather than in a separate workflow